src/service/viewport/viewport-impl.js

```diff
@@ -490,9 +508,17 @@ export class Viewport {
         offset = 0;
         break;
     }
-    const calculatedScrollTop = elementRect.top - this.paddingTop_ + offset;
-    const newScrollTop = Math.max(0, calculatedScrollTop);
-    const curScrollTop = this.getScrollTop();
+    let newScrollTop;
+    let curScrollTop;
+
+    if (this.useLayers_) {
```

Comment:
I think this should be `!this.userLayers_`, though I'd switch the order.

Reply:
Done.